src/backend/columnar/columnar_customscan.c

```diff
@@ -806,12 +935,21 @@ FilterPushdownClauses(PlannerInfo *root, RelOptInfo *rel, List *inputClauses)
 		 * there's something we should do with pseudoconstants here.
 		 */
 		if (rinfo->pseudoconstant ||
-			!bms_is_member(rel->relid, rinfo->required_relids) ||
-			!CheckPushdownClause(root, rel, rinfo->clause))
+			!bms_is_member(rel->relid, rinfo->required_relids))
 		{
 			continue;
 		}
 
+		Node *canonicalForm = eval_const_expressions(root, (Node *) rinfo->clause);
+		canonicalForm = (Node *) canonicalize_qual((Expr *) canonicalForm, false);
+		Expr *pushdownableExpr = ExtractPushdownClause(root, rel, canonicalForm);
+		if (!pushdownableExpr)
+		{
+			continue;
+		}
+
+		rinfo = copyObject(rinfo);
+		rinfo->clause = pushdownableExpr;
 		filteredClauses = lappend(filteredClauses, rinfo);
```

Comment:
I wonder should we put a different RestrictInfo for each arg of AND expression. just in case `predicate_refuted_by` can't figure out what to do with AND expressions.

```suggestion
		List *andArgList = make_ands_implicit(pushdownableExpr);
		Expr *andArg = NULL;
		foreach_ptr(andArg, andArgList)
		{
			RestrictInfo *newRestrictInfo = copyObject(rinfo);
			newRestrictInfo->clause = andArg;
			filteredClauses = lappend(filteredClauses, newRestrictInfo);
		}
```

Reply:
forget about it